### PR TITLE
Input Component

### PR DESCRIPTION
**Pull request by spiropoulos94**
This is a test commit. 
image

Is this correct?

### CODE REVIEW

**Comment by nikoskleidis** [comment]
Yes proceed

File: src/components/LoginStatus/index.js [hunk 1]
```diff
@@ -19,6 +20,9 @@ function LoginStatus(props) {
       <button onClick={() => updateLocale("en")}>English</button>
       <button onClick={() => updateLocale("el")}>Ελληνικά</button>
       {props.loggedStatus}
+      <div>
```
Comment (nikoskleidis) [inline src/components/LoginStatus/index.js:23]: You can place this in the src/Pages/TestPage.js in order to have all the components in a common place for demonstration purposes

File: src/components/Input/styles.js [hunk 2]
```diff
@@ -0,0 +1,65 @@
+import { css } from "emotion"
+
+export const InputCls = css`
```
Comment (nikoskleidis) [inline src/components/Input/styles.js:3]: Le's use lower case for the classNames

File: src/components/Input/index.js [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  disabled: PropTypes.bool,
+  errorMessage: PropTypes.string
+}
+
+function Input({ className, onChange, label, value, disabled, errorMessage }) {
+  const [inputValue, setInputValue] = useState("")
+  const [isDisabled, setIsDisabled] = useState(false)
+  const [error, setError] = useState(false)
+
+  errorMessage = "We are sorry, an error occured. Please try again."
```
Comment (nikoskleidis) [inline src/components/Input/index.js:20]: This should be removed. If you want to test the component add to the TestPage and pass the errorMessage property to the input

File: src/components/Input/index.js [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  return (
+    <div className={InputCls}>
+      <label>
+        This is an input Component
+        <input
+          type="text"
+          onChange={(e) => {
+            e.preventDefault()
+            setInputValue(e.target.value)
+            console.log(inputValue)
```
Comment (nikoskleidis) [inline src/components/Input/index.js:31]: This should be removed. No console.logs in the final commits.

File: src/components/Input/index.js [hunk 5]
```diff
@@ -0,0 +1,40 @@
...
+  className: PropTypes.string,
+  onChange: PropTypes.func,
+  label: PropTypes.string,
+  value: PropTypes.string,
+  disabled: PropTypes.bool,
+  errorMessage: PropTypes.string
+}
+
+function Input({ className, onChange, label, value, disabled, errorMessage }) {
+  const [inputValue, setInputValue] = useState("")
```
Comment (nikoskleidis) [inline src/components/Input/index.js:16]: No need to keep the value on the input in the state. The value is kept by the parent that hosts the input

File: src/components/Input/index.js [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+  const [isDisabled, setIsDisabled] = useState(false)
```
Comment (nikoskleidis) [inline src/components/Input/index.js:17]: disabled is passed as prop. No need to keep it in the state as well

File: src/components/Input/index.js [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+  const [error, setError] = useState(false)
```
Comment (nikoskleidis) [inline src/components/Input/index.js:18]: If there is an error to the field, the errorMessage prop has a value. If not then the errorMessage prop is empty. No need to keep an internal error state

File: src/components/Input/index.js [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+
+  errorMessage = "We are sorry, an error occured. Please try again."
+
+  return (
+    <div className={InputCls}>
```
Comment (nikoskleidis) [inline src/components/Input/index.js:23]: we should also add the className prop here. eg 
```suggestion
    <div className={cx(InputCls, className)}>
```

File: src/components/Input/index.js [hunk 9]
```diff
@@ -0,0 +1,40 @@
...
+      <label>
```
Comment (nikoskleidis) [inline src/components/Input/index.js:24]: I didn't see a label element in the input in the HTML pages

File: src/components/Input/index.js [hunk 10]
```diff
@@ -0,0 +1,40 @@
...
+        This is an input Component
+        <input
+          type="text"
+          onChange={(e) => {
+            e.preventDefault()
+            setInputValue(e.target.value)
+            console.log(inputValue)
+          }}
+          disabled={isDisabled}></input>
```
Comment (nikoskleidis) [inline src/components/Input/index.js:33]: The label should be added in the placeholder property of the input

File: src/components/Input/index.js (hunk 9 above)
Comment (apsuhos) [reply src/components/Input/index.js:24]: It is not. It is a different component which is used elsewhere too.

**Comment by nikoskleidis** [comment]
Closing this in favour of https://github.com/SocialHackerCreteClass/FrodidApp-FrontEnd/pull/13
The comments of this PR still apply

File: src/components/LoginStatus/index.js (hunk 1 above)
Comment (spiropoulos94) [reply src/components/LoginStatus/index.js:23]: i moved it out of the login status. should i add a new route path on App.js which will lead to the testpage? like http://localhost:3000/test-page

File: src/components/Input/index.js (hunk 3 above)
Comment (spiropoulos94) [reply src/components/Input/index.js:20]: you mean like :
`<Input errorMessage={errorMessage}/>`  ?

File: src/components/Input/index.js (hunk 5 above)
Comment (spiropoulos94) [reply src/components/Input/index.js:16]: so the value will be passed as a property from the element that contains the component?

File: src/components/Input/index.js (hunk 8 above)
Comment (spiropoulos94) [reply src/components/Input/index.js:23]: ok but why should we add a class name that is already defined above?

**Pull request closed without merging**